### ISSUE TITLE
Bad credentials after forced password change

### ISSUE THREAD

**Issue by robglas** [vmware/mangle#22]
**Describe the bug**
After the forced initial password change, Mangle doesn't accept either the old or the new password.

**To Reproduce**
1. Deploy Mangle 2.0
2. Load /mangle-services
3. Change the password from 'admin' to something else.
4. Attempt to login with the new password
5. UI returns "Invalid username or password"

**Expected behavior**
Login succeeds.

**Logs**
```
2020-01-14 20:53:19.976 [https-jsse-nio-8443-exec-6] INFO  com.vmware.mangle.services.UserService.terminateUserSession (408) - The User admin@mangle.local not Exists/logged In
2020-01-14 20:53:20.027 [https-jsse-nio-8443-exec-6] ERROR com.vmware.mangle.services.config.MangleBasicAuthenticationEntryPoint.getMangleErrorDetails (66) - Authentication for the user admin@mangle.local failed with the error: Bad credentials
```

**Configuration information:**
 - Deployment Type: Container
 - Deployment Mode: Single node
 - Client OS: OS X
 - Client Browser: Chrome 80
 
**Additional context**

**Comment by robglas** [comment]
Full snippet of a fresh auth attempt 24 hours later:
```
2020-01-15 22:27:23.554 [https-jsse-nio-8443-exec-2] ERROR com.vmware.mangle.services.config.MangleBasicAuthenticationEntryPoint.getMangleErrorDetails (66) - Authentication for the user null failed with the error: Full authentication is required to access this resource
2020-01-15 22:27:23.692 [https-jsse-nio-8443-exec-10] INFO  com.vmware.mangle.services.PasswordResetService.readResetStatus (40) - Retrieving default user password status
2020-01-15 22:27:23.949 [https-jsse-nio-8443-exec-9] INFO  com.vmware.mangle.services.ADAuthProviderService.getAllADAuthProvider (59) - Fetching all the AD authentication providers configured
2020-01-15 22:27:31.773 [https-jsse-nio-8443-exec-8] INFO  com.vmware.mangle.services.UserService.terminateUserSession (408) - The User admin@mangle.local not Exists/logged In
2020-01-15 22:27:31.819 [https-jsse-nio-8443-exec-8] ERROR com.vmware.mangle.services.config.MangleBasicAuthenticationEntryPoint.getMangleErrorDetails (66) - Authentication for the user admin@mangle.local failed with the error: Bad credentials
```

Not quite sure where to start looking for this. Any tips?

**Comment by chethanvmware** [comment]
The forced password change that one has to do during the first deployment seems to have been broken. 

As a workaround,
Can you please try using the default initial password? 

Once logged in, you can change the default password from the home page.
> Click on the username on the header in the home page
> Select "Change Password" option
> Once changed, you should be able to login with the new password

**Comment by robglas** [comment]
Since I was forced to change the default password, the default password no longer works and I'm unable to login.

**Comment by robglas** [comment]
Perfect, that did the trick. Thank you!

**Comment by suprasupa** [comment]
Is this issue already fixed in Mangle 3.0 ? If not what is the workaround. I have deployed Mangle in ECS Fargate and now not able to login neither with changed password or with the default password.
